platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/software/internal/DefaultSoftwareTypeRegistry.java

```diff
@@ -76,4 +71,43 @@ public Set<SoftwareTypeImplementation> getSoftwareTypeImplementations() {
         }
         return softwareTypeImplementations;
     }
+
+    @Override
+    public boolean isRegistered(Class<? extends Plugin<?>> pluginClass) {
+        return pluginClasses.stream().anyMatch(registered -> registered.isAssignableFrom(pluginClass));
+    }
+
+    private static class AddSoftwareTypeImplementationForPluginClass implements TypeMetadataWalker.StaticMetadataVisitor {
```

Comment:
🤔 Maybe a better name might be `SoftwareTypeRecordingMetadataVisitor`?

Reply:
Changed it to `SoftwareTypeImplementationRecordingVisitor`.  Hopefully that's clear enough.